craft_store/base_client.py

```diff
@@ -31,6 +32,54 @@
 
 logger = logging.getLogger(__name__)
 
+# The known types of credentials: a string for Candid and a Dict for UbuntuOne
+CandidCredType = str
+UbuntuOneCredType = typing.Dict[str, str]
+CredentialsType = typing.Union[CandidCredType, UbuntuOneCredType]
+
+
+def wrap_credentials(token_type: str, credentials: CredentialsType) -> str:
+    """Create a payload string that contains both `credentials` and its identifying type.
+
+    This function creates a string that contains the desired `credentials` but also
+    stores their "type", for unwrapping later with `unwrap_credentials()`.
+
+    :param token_type: The identifier for the type of token stored in `credentials`
+    :param credentials: The actual type-dependent credentials.
+    :return: A payload string ready to be passed to Auth.set_credentials()
+    """
+    return json.dumps({"t": token_type, "v": credentials})
+
+
+def unwrap_credentials(
+    token_type: str, stored_credentials: str, deserialize_old_creds: bool
+) -> CredentialsType:
+    """Retrieve the type-specific "inner" credentials from `credentials`.
+
+    This function also handles backwards-compatibility by supporting `credentials`
+    from before we stored the `token_type`; it is meant to be called with the
+    returned value from Auth.get_credentials().
+
+    :param token_type: The expected token type, for validation.
+    :param stored_credentials: The credentials retrieved from auth storage.
+    :param deserialize_old_creds:
```

Comment:
I had to add this boolean because, at least for the tests, there are two different scenarios with "old" creds that have to be handled differently:
1. Old u1 creds, so a json-encoded dict: should decode before returning
2. Old candid creds, so a string that is possibly a json value. In this case we shouldn't decode because the value is going to be passed forward (in requests) as-is. This happened in StoreClient tests that mock the credentials to be a json-encoded dict of a "plaintext macaroon", like this:

```python
@pytest.fixture
def real_macaroon():
    return json.dumps(
        {
            "s64": "a0Vi7CwhHWjS4bxzKPhCZQIEJDvlbh9FyhOtWx0tNFQ",
            "c": [
                {"i": "time-before 2022-03-18T19:54:57.151721Z"},
                {
                    "v64": "pDqaL9KDrPfCQCLDUdPc8yO2bTQheWGsM1tpxRaS_4BT3r6zpdnT5TelXz8vpjb4iUhTnc60-x5DPKJOpRuwAi4qMdNa67Vo",
                    "l": "https://api.jujucharms.com/identity/",
                    "i64": "AoZh2j7mbDQgh3oK3qMqoXKKFAnJvmOKwmDCNYHIxHqQnFLJZJUBpqoiJtqra-tyXPPMUTmfuXMgOWP7xKwTD26FBgtJBdh1mE1wt3kf0Ur_TnOzbAWQCHKxqK9jAp1jYv-LlLLAlQAmoqvz9fBf2--dIxHiLIRTThmAESAnlLZHOJ7praDmIScsLQC475a85avA",
                },
            ],
            "l": "api.snapcraft.io",
            "i64": "AwoQ2Ft5YBjnovqdr8VNV3TSlhIBMBoOCgVsb2dpbhIFbG9naW4",
        }
    )
```
... since this is a test mock I don't know if "production" apps use this, so I'm reluctant to break the behavior. Open to suggestions though!